internal/cmd/flink/command_shell.go

```diff
@@ -69,6 +73,19 @@ func (c *command) authenticated(authenticated func(*cobra.Command, []string) err
 }
 
 func (c *command) startFlinkSqlClient(prerunner pcmd.PreRunner, cmd *cobra.Command) error {
+	// if the --fake-gateway flag is set, we start the client with a simulated gateway client that returns fake data
```

Comment:
Our other commands use the integration test framework in the `test/` directory to accomplish the same thing. Have you tried this yet?

Reply:
Yeah we've looked at that. I don't think it will help us in this particular case though. We don't want to run a test case here, rather we want to open the client as usual, but instead of making real backend calls we create fake data randomly. Then you can actually check how the table view looks and how the navigation feels etc.

For integration tests though, we might be able to use your framework. I think we would have to create another fake backend that returns non-random data, so we can create the golden file more easily. Only thing I'm unsure about is, how we can issue statements inside the client. I saw that you are using `exec.Command` to run the commands, but I'm not sure if this also works once you are inside the client, and want to write something into the go-prompt buffer.

Reply:
We need this here for faster development, not for testing. Devel/stag is often unstable, either not available or slow and we're often working on features (like coloring), where we want to start the client with a mocked backend so we test stuff.